src/coreclr/src/System.Private.CoreLib/src/System/StartupHookProvider.cs

```diff
@@ -102,6 +103,8 @@ private static void ProcessStartupHooks()
 
         // Load the specified assembly, and call the specified type's
         // "static void Initialize()" method.
+        [RequiresUnreferencedCode("Trimming may remove assemblies, types or members used by startup hooks. " +
+                                  "These can be disabled using the StartupHookSupport feature switch")]
```

Comment:
@samsp-msft can you help us here with the message wording?

This is the message for a warning which will show up if an app allows startup hooks but asks for trimming. Since startup hooks by design load "random" code into the process, that code may have dependencies on things which were trimmed and thus it won't work.

The goal of the message should be:
* Describe what the problem is - short version "startup hooks are not compatible with trimming"
* At least hint at possible solution - in this case use the StartupHookSupport feature switch to disable startup hooks.

My attempt:
"This application enables startup hooks which might require assemblies, types or members removed during trimming. Startup hooks can be disabled with the StartupHookSupport feature switch."

Reply:
Potential Incompatibility: The StartupHookSupport feature switch has been enabled for this app which is being trimmed. Startup hook code is not observable by the trimmer and so required assemblies, types and members may be removed.